compiler/generator/csharp/csharp_instructions.hh

```diff
@@ -248,7 +244,23 @@ class CSharpInstVisitor : public TextInstVisitor {
             *fOut << fTypeManager->generateType(inst->fType, inst->fAddress->getName());
             if (inst->fValue) {
                 *fOut << " = ";
-                inst->fValue->accept(this);
+
+                if (dynamic_cast<BinopInst*>(inst->fValue)) {
```

Comment:
This code is required to handle temporary variable assignments where a boolean value coming from a binary operation is being assigned to an integer temporary variable.

I had to do the dynamic_cast check for BinopInst because if I didn't, I was getting errors from the TypingVisitor check for certain kinds of instructions. The bool/int conversion needs to be done for any boolean expression, but I think that BinopInst is the only situation where boolean values occur.